site/app/controllers/admin/NotebookBuilderController.php

```diff
@@ -26,6 +29,30 @@ public function __construct(\app\libraries\Core $core) {
         $this->expected_group = $this->core->getConfig()->getCourse() . '_tas_www';
     }
 
+    /**
+     * @param string $g_id Gradeable ID
+     * notebook builder know to save a new configuration or edit the existing one.
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    #[Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}/notebook_preview", methods: ["GET"])]
+    public function preview(string $g_id) {
+        $gradeable = $this->core->getQueries()->getGradeableConfig($g_id);
+        $file_path = $gradeable->getAutogradingConfigPath() . '/config.json';
+        $filedata = FileUtils::readJsonFile($file_path);
+
+        $this->core->getOutput()->addInternalCss('notebook-preview.css');
```

Comment:
This file isn't included in the changes you've pushed thus far.  Did you forget to commit it, or is this now unnecessary?  I suspect that it will be unnecessary once you switch to sharing common Twig code.

Reply:
Previously, I was adding custom css but it wasn’t necessary so didn’t commit it.